File: src/ruby/Hanoi.rb

def Hanoi(pin0, pin2, pin1, num)
   if num == 1
   		# Show the operations
       print "Move from ", pin0, " to ", pin2, "\n"
   else
       Hanoi(pin0, pin1, pin2, num-1)
       Hanoi(pin0, pin2, pin1, 1)
       Hanoi(pin1, pin2, pin0, num-1)
   end
end

# Move from Pin-0 to Pin-2 using Pin-1 as helper (number of disks = 3)
Hanoi(0, 2, 1, 3)

File: src/swift/hanoi.swift
//
//  hanoi.swift
//
//  Created by Matheus Torres on 13/11/20.
//

func hanoi(_ p1: Int, _ p2: Int, _ p3: Int, disks: Int) {
    if disks == 1 {
        print("Move de \(p1) para \(p2)")
    } else {
        hanoi(p1, p3, p2, disks: disks - 1)
        hanoi(p1, p2, p3, disks: 1)
        hanoi(p3, p2, p1, disks: disks - 1)
    }
}

hanoi(0, 2, 1, disks: 3)

